components/renku_data_services/notebooks/config/dynamic.py

```diff
@@ -456,8 +461,8 @@ def _for_testing(cls) -> Self:
         )
 
     @property
-    def affinity_model(self) -> Affinity:
-        return Affinity.model_validate(self.affinity)
+    def affinity_model(self) -> Affinity | None:
+        return Affinity.model_validate(self.affinity) if self.affinity is not None else None
```

Comment:
Why not return the empty `Affinity()` instance when there is no specific affinity configured? This way you do not have to update all the types in this commit.